.github/actions/python-build/action.yml

```diff
@@ -0,0 +1,45 @@
+name: "Build the python code and wheels"
+description: "Common Python build steps"
+inputs:
+  type:
+    description: python_version
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/setup-python@v4
+      name: Install Python
+      with:
+        python-version: ${{ inputs.python_version }}
+
+    - uses: Gr1N/setup-poetry@v8
```

Comment:
Any reasons for this poetry setup action vs the one in our other repos or something like https://github.com/snok/install-poetry? We should turn on caching (fine to do in follow up).

Reply:
No reason at all, I just happened to find this one first and I've been using it. Looks like they take the same options more or less so I can sub in the other one.